The code check complains about a missing header,

```
/.../CMSSW_10_1_X_2018-02-17-1100/src/PhysicsTools/TensorFlow/src/NTSession.h:49:10: error:
'tensorflow/core/common_runtime/graph_execution_state.h' file not found
```

which is added by cms-sw/cmsdist#3717. @slava77 @smuzaffar Is there anything I can do to make this PR consider cms-sw/cmsdist#3717 for code checks and tests?

--- a/PhysicsTools/TensorFlow/src/NTSession.cc
+++ b/PhysicsTools/TensorFlow/src/NTSession.cc
@@ -12,6 +12,9 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
+//NOTE: The memory layout of the Node class changes depending on if NDEBUG was
+// set when tensorflow was compiled. The reason is Node class holds two edgeset
+// class instances and edgeset adds a member data if NDEBUG is set
 
 /*
 This file is an adaptation of the original direct_session.cc file located at
@@ -171,6 +174,21 @@ static NTSessionRegistrar registrar;
 
 std::atomic_int_fast64_t NTSession::step_id_counter_(1);
 
+// NOTE: On Android with a single device, there is never
+// a risk of an OpKernel blocking indefinitely:
+//
+// 1) No operations do I/O that depends on other simultaneous kernels,
+//
+// 2) Recv nodes always complete immediately: The inputs are sent into
+//    the local rendezvous before we start the executor, so the
+//    corresponding recvs will not block.
+//
+// Based on these assumptions, we can use the same thread pool for
+// both "non-blocking" and "blocking" OpKernels on Android.
+//
+// This may change down the road when we add support for multiple
+// devices that run concurrently, in which case we will need to
+// revisit this decision.
 // Override to allow CMSSW FWK to schedule
 void NTSession::SchedClosure(std::function<void()> c) {
   c();
@@ -419,9 +437,7 @@ Status NTSession::Run(const RunOptions& run_options,
 
   args.rendezvous = run_state.rendez;
   args.cancellation_manager = &step_cancellation_manager;
-  args.runner = [this](Executor::Args::Closure c) {
-    SchedClosure(std::move(c));
-  };
+
   args.session_state = &session_state_;
   args.tensor_store = &run_state.tensor_store;
   args.step_container = &run_state.step_container;
@@ -482,6 +498,9 @@ Status NTSession::Run(const RunOptions& run_options,
     return errors::Cancelled("Run call was cancelled");
   }
 
+  args.runner = [this](Executor::Args::Closure c) {
+    SchedClosure(std::move(c));
+  };
   for (const auto& item : executors_and_keys->items) {
     item.executor->RunAsync(args, barrier->Get());
   }

--- a/PhysicsTools/TensorFlow/src/TBBSession.cc
+++ b/PhysicsTools/TensorFlow/src/TBBSession.cc
@@ -12,6 +12,9 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
+//NOTE: The memory layout of the Node class changes depending on if NDEBUG was
+// set when tensorflow was compiled. The reason is Node class holds two edgeset
+// class instances and edgeset adds a member data if NDEBUG is set
 
 /*
 This file is an adaptation of the original direct_session.cc file located at
@@ -187,6 +190,7 @@ std::atomic_int_fast64_t TBBSession::step_id_counter_(1);
 // This may change down the road when we add support for multiple
 // devices that run concurrently, in which case we will need to
 // revisit this decision.
+// Override to allow CMSSW FWK to schedule
 void TBBSession::SchedClosure(tbb::task_arena& arena, tbb::task_group& g, std::function<void()> c) {
   arena.execute( [&g,&c] () {g.run( c ); } );
 }
@@ -442,9 +446,7 @@ Status TBBSession::Run(const RunOptions& run_options,
 
   args.rendezvous = run_state.rendez;
   args.cancellation_manager = &step_cancellation_manager;
-  args.runner = [this, &taskArena, &taskGroup](Executor::Args::Closure c) {
-    SchedClosure(taskArena, taskGroup, std::move(c));
-  };
+
   args.session_state = &session_state_;
   args.tensor_store = &run_state.tensor_store;
   args.step_container = &run_state.step_container;
@@ -505,6 +507,9 @@ Status TBBSession::Run(const RunOptions& run_options,
     return errors::Cancelled("Run call was cancelled");
   }
 
+  args.runner = [this, &taskArena, &taskGroup](Executor::Args::Closure c) {
+    SchedClosure(taskArena, taskGroup, std::move(c));
+  };
   for (const auto& item : executors_and_keys->items) {
     item.executor->RunAsync(args, barrier->Get());
   }

--- a/PhysicsTools/TensorFlow/src/TBBSession.cc
+++ b/PhysicsTools/TensorFlow/src/TBBSession.cc
@@ -357,7 +357,7 @@ Status TBBSession::Run(const RunOptions& run_options,
                           std::vector<Tensor>* outputs,
                           RunMetadata* run_metadata) {
   TF_RETURN_IF_ERROR(CheckNotClosed());
-  direct_session_runs->GetCell()->IncrementBy(1);
+  tbb_session_runs->GetCell()->IncrementBy(1);
   {
     mutex_lock l(graph_def_lock_);
     if (!graph_created_) {

